staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go

```diff
@@ -80,6 +81,24 @@ func createHandler(r rest.NamedCreater, scope RequestScope, admit admission.Inte
 			return
 		}
 
+		options := &metav1.CreateOptions{}
+		if values := req.URL.Query(); len(values) > 0 {
+			// TODO: replace with content type negotiation?
+			// "1" is accepted as "true" for backward compatibility.
+			for i, v := range values["includeUninitialized"] {
+				if v == "1" {
+					values["includeUninitialized"][i] = "true"
+				}
+			}
+			// Timeout is not part of the options, drop it.
+			delete(values, "timeout")
```

Comment:
DecodeParameters ignores params that are not in the struct (necessary for new clients speaking to old servers). So, no need for the delete.

Reply:
Fixed